beprof/curve.py

```diff
@@ -75,8 +78,21 @@ def y(self):
     def y(self, value):
         self[:, 1] = value
 
-    def rescale(self, factor=1.0):
-        self.y /= factor
+    def rescale(self, factor=1.0, allow_cast=True):
+        """
+        todo: write docstring nad doctests
+        :param factor:
+        :param allow_cast:
+        :return:
+        """
+        try:
+            self.y /= factor
+        except TypeError as e:
+            logger.error("Division in place is impossible.\n%s" % e)
+            if allow_cast:
+                self.y = self.y / factor
+            else:
+                logger.error("allow_cast flag set to True should help")
```

Comment:
It stops, we could go for re-raise at the end - after calculating with ` self.y = self.y / factor` or ignore TypeErrors thrown by  `self.y /= factor` or... just log it?
```py
        try:
            self.y /= factor
        except TypeError as e:
            logger.error("Division in place is impossible")
            if allow_cast:
                self.y = self.y / factor
            else:
                logger.error("allow_cast flag set to True should help")
            raise e
```